addons/web_tour/static/src/js/tip.js

```diff
@@ -156,41 +235,76 @@ var Tip = Widget.extend({
+        }
 
         // Reverse overlay if direction is right to left
         var positionRight = _t.database.parameters.direction === 'rtl' ? "right" : "left";
         var positionLeft = _t.database.parameters.direction === 'rtl' ? "left" : "right";
-        var offset = this.$el.offset();
+
+        // get the offset position of this.$el
+        // Couldn't use offset() or position() because their values are not the desired ones in all cases
+        const offset = {top: this.$el[0].offsetTop, left: this.$el[0].offsetLeft};
         this.$tooltip_overlay.css({
             top: -Math.min((this.info.position === "bottom" ? this.info.space : this.info.overlay.y), offset.top),
             right: -Math.min((this.info.position === positionRight ? this.info.space : this.info.overlay.x), this.$window.width() - (offset.left + this.init_width + this.double_border_width)),
             bottom: -Math.min((this.info.position === "top" ? this.info.space : this.info.overlay.y), this.$window.height() - (offset.top + this.init_height + this.double_border_width)),
             left: -Math.min((this.info.position === positionLeft ? this.info.space : this.info.overlay.x), offset.left),
         });
-
-        this.position = this.$el.position();
+        this.position = offset;
 
         this.$el.addClass("o_animated");
     },
     _bind_anchor_events: function () {
         this.consume_event = Tip.getConsumeEventType(this.$anchor);
+        if (this.info.consumeAction === "input") {
+            var contentEditable = this.$anchor.closest("[contenteditable='true']");
+            if (this.consume_event !== this.info.consumeAction && contentEditable.length) {
+                this.consume_event = "contenteditable";
+            }
+        }
         this.$consumeEventAnchor = this.$anchor;
         // jQuery-ui draggable triggers 'drag' events on the .ui-draggable element,
         // but the tip is attached to the .ui-draggable-handle element which may
         // be one of its children (or the element itself)
         if (this.consume_event === "drag") {
             this.$consumeEventAnchor = this.$anchor.closest('.ui-draggable');
+        } else if (this.consume_event === "contenteditable") {
+            this.$consumeEventAnchor = contentEditable;
+            this.consume_event = "input";
         } else if (this.consume_event.includes('apply.daterangepicker')) {
             this.$consumeEventAnchor = this.$anchor.parent().children('.o_field_date_range');
         }
         this.$consumeEventAnchor.on(this.consume_event + ".anchor", (function (e) {
-            if (e.type !== "mousedown" || e.which === 1) { // only left click
+            if (e.type === "input") {
+                this.trigger("tip_consumed");
+                this._unbind_anchor_events();
+            } else if (e.type !== "mousedown" || e.which === 1) { // only left click
                 this.trigger("tip_consumed");
                 this._unbind_anchor_events();
             }
```

Comment:
Seriously? If A, do(), else, do()?

Reply:
just forgot to remove it, I did it for debug purpose